internal/commands/app/create_inputs.go

```diff
@@ -153,56 +154,69 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
-func (i *createInputs) resolveCluster(client atlas.Client, groupID string) (dataSourceCluster, error) {
+func (i *createInputs) resolveClusters(client atlas.Client, groupID string) ([]dataSourceCluster, error) {
 	clusters, err := client.Clusters(groupID)
 	if err != nil {
-		return dataSourceCluster{}, err
+		return []dataSourceCluster{}, err
 	}
-	var clusterName string
-	for _, cluster := range clusters {
-		if i.Cluster == cluster.Name {
-			clusterName = cluster.Name
-			break
-		}
+
+	existingClusters := map[string]atlas.Cluster{}
+	for _, c := range clusters {
+		existingClusters[c.Name] = c
 	}
-	if clusterName == "" {
-		return dataSourceCluster{}, errors.New("failed to find Atlas cluster")
+
+	var dsClusters []dataSourceCluster
+	for _, clusterName := range i.Clusters {
+		if _, ok := existingClusters[clusterName]; !ok {
+			continue
+		}
+		dsClusters = append(dsClusters,
+			dataSourceCluster{
+				Name: "mongodb-atlas",
+				Type: "mongodb-atlas",
+				Config: configCluster{
+					ClusterName:         clusterName,
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+			})
 	}
-	dsCluster := dataSourceCluster{
-		Name: "mongodb-atlas",
-		Type: "mongodb-atlas",
-		Config: configCluster{
-			ClusterName:         clusterName,
-			ReadPreference:      "primary",
-			WireProtocolEnabled: false,
-		},
+	if len(dsClusters) == 0 {
+		return []dataSourceCluster{}, errors.New("failed to find Atlas cluster")
```

Comment:
i think we want to tweak this logic a bit... previously when we only allowed one cluster, we would return this error if we failed to match that name from the returned list of clusters.

preserving that behavior would entail returning this error where we `continue` above (which would only tell the user about the first cluster not found, not potentially all of them not found).  so maybe we want to extend the error to find all missing first and communicate that back to the user.

my last idea here is: we should check with product to see if this warrants a "failure" at all.  perhaps we want to collect any missing specified cluster/datalake names and print that out to the user as a warning.  I can help follow up on that with you, but feel free to start a thread with drew/sumedha on that one